rust/cubesql/cubesql/src/compile/engine/df/scan.rs

```diff
@@ -446,8 +446,11 @@ struct CubeScanExecutionPlan {
 }
 
 #[derive(Debug)]
-pub enum FieldValue {
+pub enum FieldValue<'a> {
+    // Neon doesn't allow us to build string reference, because V8 uses UTF-16, while Rust uses UTF-8
     String(String),
+    // Best variant for us with strings, because we dont need to clone string
+    StringRef(&'a String),
```

Comment:
Why not reference to `str`? I don't think reference to `String` gets us anything functionally, but ties to specific allocation pattern.

Reply:
`'str` cannot be used because I combine branches via `|`. In Rust, it is a requirement to align them to the same type for usage.